docs/REVIEWING.md

```diff
@@ -36,6 +36,40 @@ following items should also be checked:
 - [ ] Does it comply with [component spec](specs/component.md)?
 - [ ] Does it comply with the [instrumentation spec](specs/instrumentation.md)?
 
+
+### Sink review checklist
+
+This checklist is specific for Vector's sink code.
+
+## Logic
+
+- [ ] Does it work? Do you understand what it is supposed to be doing?
+- [ ] Does the retry logic make sense?
+- [ ] Are the tests testing that the sink is emitting the correct metrics?
```

Comment:
Just a note that when the component validation framework is done, we could make this more specific, even link to a separate docs that describe how to add the tests for that.